File: 453/453-minMoves.go
// https://leetcode-cn.com/problems/minimum-moves-to-equal-array-elements/
package main

import "fmt"

func minMoves(nums []int) int {
	min := nums[0]
	for _, v := range nums {
		if min > v {
			min = v
		}
	}
	count := 0
	for _, v := range nums {
		count += v - min
	}
	return count
}

func main() {
	n := [][]int{{1, 2, 3}, {0, 2, 4}, {1, 4, 4}}
	for _, v := range n {
		fmt.Println("ret:", minMoves(v))
	}
}

File: 201-500/453/rust/453-min_moves.rs
// https://leetcode-cn.com/problems/minimum-moves-to-equal-array-elements/

impl Solution {
    pub fn min_moves(nums: Vec<i32>) -> i32 {
        let mut min = nums[0];
        for num in &nums {
            if min > *num {
                min = *num;
            }
        }
        let mut count = 0;
        for num in &nums {
            count += *num - min;
        }
        return count;
    }
}

pub struct Solution {}

fn main() {
    let num = vec![1, 2, 3];
    println!("ret:{}", Solution::min_moves(num));
}
